doc/src/manual/strings.md

```diff
@@ -565,14 +565,15 @@ Some other useful functions include:
 
   * [`endof(str)`](@ref) gives the maximal (byte) index that can be used to index into `str`.
   * [`length(str)`](@ref) the number of characters in `str`.
+  * [`length(str, i, j)`](@ref) the number of valid character indices in `str` from `i` to `j`.
```

Comment:
Actually I wonder whether we should add this method, as it looks a bit weird. Would it be enough to recommend `length(SubString(str, i:j))`?

Reply:
It likely can be free if it gets inlined. But seem to recall the SubString constructor does quite a bit of work to munge the indices into something valid? Anyways, doesn't seem like this API decision should be too much influenced by temporary performance concerns (unless it's only for internal usage, but them maybe call it `Base.substr_length`).